Source/Core/InputCommon/ControllerInterface/Xlib/XInput2.cpp

```diff
@@ -312,6 +340,14 @@ void KeyboardMouse::UpdateInput()
         if (delta_delta == delta_delta && 1 + delta_delta != delta_delta)
           delta_y += delta_delta;
       }
+      if (XIMaskIsSet(raw_event->valuators.mask, 3))
+      {
+        // Scroll wheel input gets scaled to be similar to the mouse axes
+        delta_delta = raw_event->raw_values[0] * 8.0 / scroll_increment;
```

Comment:
I was unable to find good documentation on this, but I assume you tested this and it works as expected? I find it a bit odd that `raw_event->raw_values[0]` is used here when I would expect `raw_values[3]` based on the other two branches here...

Reply:
Searching around a bit more, I think our handling of `raw_values` and `valuators.mask` is wrong in general; based on https://people.freedesktop.org/~whot/xi2-recipes/part4.c it's supposed to be used like this:

```
size_t value_idx = 0;
for (int i = 0; i < event->valuators.mask_len * 8; ++i) {
	if (XIMaskIsSet(event->valuators.mask, i)) {
		const double value = event->raw_values[value_idx];
		// use value here
		++value_idx;
	}
}
```

I guess it just happens to work as-is because bit 1 is never set when bit 0 is clear...?

Reply:
Yeah, XInput2 (as with a lot of X11 stuff) is not very well documented. I think you're correct here. In practice, what already exists was mostly fine, since (from what I've tested), events for mouse X and Y, and events for scroll axes are always sent separately, but that's mostly an implementation detail I think. It shouldn't be much trouble to change it